src/components/smartSearch/EditTargetDialog.tsx

```diff
@@ -0,0 +1,128 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { useState } from 'react';
+import { Box, Button, ButtonBase, Card, CardContent, Dialog, DialogContent, Typography } from '@material-ui/core';
+
+import All from './filters/All';
+import Filter from './Filter';
+import { isFilterWithId } from './utils';
+import MostActive from './filters/MostActive';
+import patchTaskTarget from 'fetching/tasks/patchTaskTarget';
+import { useRouter } from 'next/router';
+import { ZetkinSmartSearchFilter } from 'types/zetkin';
+import { FILTER_TYPE, ZetkinSmartSearchFilterWithId } from 'types/smartSearch';
+import { useMutation, useQueryClient } from 'react-query';
+
+interface EditTargetDialogProps {
+    filterSpec: ZetkinSmartSearchFilter[];
+    onDialogClose: () => void;
+    open: boolean;
+}
+
+const EditTargetDialog = ({ onDialogClose, open, filterSpec }: EditTargetDialogProps) : JSX.Element => {
+    const queryClient = useQueryClient();
+    const { orgId, taskId } = useRouter().query;
+    const filtersWithIds = filterSpec.map((filter, index) => ({ ...filter, id: index }));
+    const [filterArray, setFilterArray] = useState<ZetkinSmartSearchFilterWithId[]>(filtersWithIds);
+    const [selectedType, setSelectedType] = useState<FILTER_TYPE | null>(null);
+    const [selectedFilter, setSelectedFilter] = useState<ZetkinSmartSearchFilterWithId| null>(null);
+
+    const taskMutation = useMutation(patchTaskTarget(orgId as string, taskId as string),{
+        onSettled: () => queryClient.invalidateQueries(['filter_spec', orgId, taskId]),
```

Comment:
We need to think about how we name our cache keys. This is a Smart Search query and it has it's own ID, no need to reference the task ID (I don't hink, unless that's required by something else).

Reply:
I also realised that since i'm fetching `/orgs/orgId/tasks/taskId`, it's allowing me to access the task target at `campaigns/[wrongcampaignID]/tasks/assignees`.

Reply:
Changed so that now I'm actually invalidating the ['task', orgId, taskId] query, since that's where we're getting the task target from in the first place. When I patch, i update the 'query' object directly, but still invalidate the task query, isn't this what I should be doing here?